projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php

```diff
@@ -1178,7 +1180,19 @@ function property_is_valid($property) {
 		$property = strtolower($property);
 		if (in_array(trim($property), $GLOBALS['csstidy']['multiple_properties'])) $property = trim($property);
 		$all_properties = & $GLOBALS['csstidy']['all_properties'];
-		return (isset($all_properties[$property]) && strpos($all_properties[$property], strtoupper($this->get_cfg('css_level'))) !== false );
+		return ((isset($all_properties[$property]) && strpos($all_properties[$property], strtoupper($this->get_cfg('css_level'))) !== false) 
+					|| ($this->get_cfg('preserve_css_variables') && $this->property_is_css_variable($property)));
+	}
+
+	/**
+	 * Checks if a property is a css variable
+	 * @param string $property
+	 * @return bool;
+	 * @access public
+	 * @version 1.0
+	 */
+	function property_is_css_variable($property) {
+		return preg_match('/^--[a-zA-Z0-9\-_]*/', $property);
```

Comment:
It might be handy to add a note in the comments as to what a valid variable pattern is, e.g., 

`Valid patterns must start with `--` and use lowercase alphanumeric characters optionally separated by `-`. They must not end with a `-`.`

Unless there are example of CSS vars using underscores and uppercase, we could optionally enforce some sort of a standard of here, e.g., `--test-var-name`. 

This would tighten up the regex by disallowing more chars, `_0-9A-Z`.

So the regex might be something like `/^--([a-z]+\-?)+(?<!\-)$/` 

This allows: 

```
--var
--var-var
--var-var-var // etc
```

**But not**

```
--
--var-
--var0
--VAR-n
```

Or with alphanumeric (lowercase): `/^--([a-z0-9]+\-?)+(?<!\-)$/`

Reply:
Thanks for pointing me to this: https://www.w3.org/TR/CSS22/syndata.html#value-def-identifier

Adding valid chars would just look like this: `^--([a-zA-Z0-9]+\-?)+(?<!\-)$`

I think the current regex still lets through `--`, with no var names.